pkg/frontend/querymiddleware/codec.go

```diff
@@ -112,7 +116,45 @@ type Response interface {
 	GetHeaders() []*PrometheusResponseHeader
 }
 
-type prometheusCodec struct{}
+type prometheusCodecMetrics struct {
+	duration *prometheus.HistogramVec
+	size     *prometheus.HistogramVec
+}
+
+func newPrometheusCodecMetrics(registerer prometheus.Registerer) *prometheusCodecMetrics {
+	factory := promauto.With(registerer)
+	ms := 1.0 / 1000
+	us := ms / 1000
+	kb := 1024.0
+	mb := 1024 * kb
+
+	return &prometheusCodecMetrics{
+		duration: factory.NewHistogramVec(prometheus.HistogramOpts{
+			Namespace: "cortex",
+			Name:      "frontend_query_payload_codec_duration_seconds",
```

Comment:
[nit] Two small suggestions:
- Put the whole metric name in Name instead of using Namespace. Makes "grep-ing" the metric name working.
- WDYT about using "response" instead of "payload"? Looks easier to understand to me.

_If you accept these comments, then they apply to other metrics below._

```suggestion
			Name:      "cortex_frontend_query_response_codec_duration_seconds",
```

Reply:
Makes sense, resolved in 38d4b84.